pkg/porter/credentials.go

```diff
@@ -2,16 +2,100 @@ package porter
 
 import (
 	"fmt"
+	"os"
+	"sort"
+	"time"
 
 	"github.com/deislabs/porter/pkg/context"
-	"github.com/deislabs/porter/pkg/credentials"
+	"github.com/deislabs/porter/pkg/credentialsgenerator"
 	"github.com/deislabs/porter/pkg/printer"
+
+	dtprinter "github.com/carolynvs/datetime-printer"
+	credentials "github.com/deislabs/cnab-go/credentials"
 	"github.com/pkg/errors"
 	yaml "gopkg.in/yaml.v2"
 )
 
-func (p *Porter) PrintCredentials(opts printer.PrintOptions) error {
-	return nil
+// CredentialsFile represents a CNAB credentials file and corresponding metadata
+type CredentialsFile struct {
+	Name     string
+	Modified time.Time
+}
+
+// CredentialsFileList is a slice of CredentialsFiles
+type CredentialsFileList []CredentialsFile
+
+func (l CredentialsFileList) Len() int {
+	return len(l)
+}
+func (l CredentialsFileList) Swap(i, j int) {
+	l[i], l[j] = l[j], l[i]
+}
+func (l CredentialsFileList) Less(i, j int) bool {
+	return l[i].Modified.Before(l[j].Modified)
+}
+
+// ListCredentials lists credentials using the provided printer.PrintOptions
+func (p *Porter) ListCredentials(opts printer.PrintOptions) error {
+	dir, err := p.Config.GetCredentialsDir()
+	if err != nil {
+		return errors.Wrap(err, "unable to determine credentials directory")
+	}
+
+	// TODO: should this go somewhere else?
+	// i.e., where should logic for init-ing dirs needed by porter go?
+	// e.g., on 'porter creds list' with a fresh porter install/home
+	if ok, _ := p.Context.FileSystem.DirExists(dir); !ok {
+		p.Context.FileSystem.Mkdir(dir, os.ModePerm)
+	}
+
+	credentialsFiles := CredentialsFileList{}
+	err = p.Context.FileSystem.Walk(dir, func(path string, info os.FileInfo, err error) error {
+		if !info.IsDir() {
+			credSet := &credentials.CredentialSet{}
+			data, err := p.Context.FileSystem.ReadFile(path)
+			if err != nil {
+				return errors.Wrap(err, fmt.Sprintf("unable to load credential set from %s:\n%s", path, err))
+			}
+			if err = yaml.Unmarshal(data, credSet); err != nil {
+				return errors.Wrap(err, "unable to unmarshal credential set")
+			}
+			credentialsFiles = append(credentialsFiles,
+				CredentialsFile{Name: credSet.Name, Modified: info.ModTime()})
+		}
+		return nil
+	})
+	if err != nil {
+		return errors.Wrap(err, "encountered error while listing credentials")
+	}
+	sort.Sort(sort.Reverse(credentialsFiles))
+
+	switch opts.Format {
+	case printer.FormatJson:
+		return printer.PrintJson(p.Out, credentialsFiles)
+	case printer.FormatYaml:
+		return printer.PrintYaml(p.Out, credentialsFiles)
+	case printer.FormatTable:
+		// TODO: update table printing
```

Comment:
I'm not clear on what this TODO is talking about?

Reply:
This TODO was a note to remind that we intend to move to a different table printing method, as mentioned in https://github.com/deislabs/porter/issues/371.  I've removed the note for now, since we're tracking in that separate issue.